### PR TITLE
DT-630: Add in user contacts endpoint

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: src/main/java/uk/gov/justice/digital/hmpps/oauth2server/resource/api/UserContactsController.kt [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  @GetMapping("/api/user/{username}/contacts")
+  @PreAuthorize("hasRole('ROLE_RETRIEVE_OAUTH_CONTACTS')")
+  @ApiOperation(value = "Get contacts for user.", notes = "Get verified contacts for user.", nickname = "contacts", consumes = "application/json", produces = "application/json")
+  @ApiResponses(value = [
+    ApiResponse(code = 200, message = "OK", response = ContactDto::class, responseContainer = "List"),
+    ApiResponse(code = 401, message = "Unauthorized.", response = ErrorDetail::class),
+    ApiResponse(code = 404, message = "User not found.", response = ErrorDetail::class)])
+  fun contacts(@ApiParam(value = "The username of the user.", required = true) @PathVariable username: String): List<ContactDto> {
+    val user = userService.getUserWithContacts(username)
+    return user.contacts.filter { it.verified }.map { ContactDto(it.value!!, it.type.name, it.type.description) }
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/digital/hmpps/oauth2server/resource/api/UserContactsController.kt]: only makes sense to expose verified contacts

File: src/main/java/uk/gov/justice/digital/hmpps/oauth2server/resource/api/UserContactsController.kt [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import uk.gov.justice.digital.hmpps.oauth2server.model.ErrorDetail
+import uk.gov.justice.digital.hmpps.oauth2server.security.UserService
+
+@Slf4j
+@RestController
+@Api(tags = ["/api/user/{username}/contacts"])
+@AllArgsConstructor
+class UserContactsController(private val userService: UserService) {
+  @GetMapping("/api/user/{username}/contacts")
+  @PreAuthorize("hasRole('ROLE_RETRIEVE_OAUTH_CONTACTS')")
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/digital/hmpps/oauth2server/resource/api/UserContactsController.kt]: new role

File: src/test/java/uk/gov/justice/digital/hmpps/oauth2server/resource/api/AuthUserIntTest.kt [hunk 3]
```diff
@@ -28,10 +27,10 @@ class AuthUserIntTest : IntegrationTest() {
         .exchange()
         .expectStatus().isOk
         .expectHeader().contentType(MediaType.APPLICATION_JSON)
-        .expectBody(object : ParameterizedTypeReference<Map<String, String>>() {})
-        .consumeWith<Nothing?> {
-          assertThat(it!!.responseBody!!.filter { it.key != "userId" }).containsExactlyEntriesOf(
-              mapOf("username" to username.toUpperCase(), "active" to "true", "name" to "Bob Smith", "authSource" to "auth"))
+        .expectBody()
+        .jsonPath("$").value<Map<String, Any>> {
```
Comment (petergphillips) [inline src/test/java/uk/gov/justice/digital/hmpps/oauth2server/resource/api/AuthUserIntTest.kt]: simpler comparison after previous PR - not sure if better way to do the `$` though - @mikehalmamoj

Comment (mikehalmamoj) [reply src/test/java/uk/gov/justice/digital/hmpps/oauth2server/resource/api/AuthUserIntTest.kt]: I don't think so.  It reads ok to me though - extract a map from the root of the json.

**Pull request merged**